File: sort/oddeven.go
package sort

<<<<<<< HEAD
<<<<<<< HEAD
//Odd-Even Sort is a variation of bubble sort where the sorting is divided into
//two phases, Odd and Even Phase and it runs until all the elements are sorted
//In the odd phase we perform bubble sort on odd indexed elements and in the
=======
//Odd-Even Sort is a variation of bubble sort where the sorting is divided into 
//two phases, Odd and Even Phase and it runs until all the elements are sorted
//In the odd phase we perform bubble sort on odd indexed elements and in the 
>>>>>>> f65a9f9... few updates
=======
//Odd-Even Sort is a variation of bubble sort where the sorting is divided into
//two phases, Odd and Even Phase and it runs until all the elements are sorted
//In the odd phase we perform bubble sort on odd indexed elements and in the
>>>>>>> 2d960f3... Deleted half implemented counting sort and formatted Odd-Even Sort
//even phase we perform bubble sort on even indexed elements.

func OddEvenSort(data []int) {
	sorted := false
	dataLen := len(data)

	for {
		sorted = true
<<<<<<< HEAD
		for i := 1; i < dataLen-1; i += 2 {
=======
		for i := 1; i < dataLen-1; i+=2 {
>>>>>>> 1f0f09e... Update oddeven.go
			if data[i] > data[i+1] {
				data[i], data[i+1] = data[i+1], data[i]
				sorted = false
			}
		}
<<<<<<< HEAD
		for i := 0; i < dataLen-1; i += 2 {
=======
		for i := 0; i < dataLen-1; i+=2 {
>>>>>>> 1f0f09e... Update oddeven.go
			if data[i] > data[i+1] {
				data[i], data[i+1] = data[i+1], data[i]
				sorted = false
			}
		}
		if sorted {
<<<<<<< HEAD
<<<<<<< HEAD
			break
=======
			break		
>>>>>>> f65a9f9... few updates
=======
			break
>>>>>>> 2d960f3... Deleted half implemented counting sort and formatted Odd-Even Sort
		}
	}
}

File: sort/oddeven.py
""" Implementation of Odd-Even Sort algorithm
"""


def oddeven(data):
    """ OddEvenSort is a variation of bubble sort where the sorting is divided
    into two phases, Odd and Even Phase and it runs until all the elements
    are sorted. In the odd phase we perform bubble sort on odd indexed
    elements and in the even phase we perform bubble sort on even indexed
    elements.

    :param array: list of elements that needs to be sorted
    :type array: list
    """
    is_sorted = False
    data_len = len(data)

    while not is_sorted:
        is_sorted = True
        for i in range(1, data_len-1, 2):
            if data[i] > data[i+1]:
                data[i], data[i+1] = data[i+1], data[i]
                is_sorted = False

        for i in range(0, data_len-1, 2):
            if data[i] > data[i+1]:
                data[i], data[i+1] = data[i+1], data[i]
                is_sorted = False


def main():
    """ operational function """
    arr = [34, 56, 23, 67, 3, 68]
    print(f"unsorted array: {arr}")
    oddeven(arr)
    print(f"  sorted array: {arr}")


if __name__ == "__main__":
    main()
